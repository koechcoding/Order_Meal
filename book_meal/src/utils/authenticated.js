export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}