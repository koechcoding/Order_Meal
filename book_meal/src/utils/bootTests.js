import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
