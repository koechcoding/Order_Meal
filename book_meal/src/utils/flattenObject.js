export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}