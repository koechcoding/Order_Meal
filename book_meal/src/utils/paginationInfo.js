export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}