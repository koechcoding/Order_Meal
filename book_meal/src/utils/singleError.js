const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;