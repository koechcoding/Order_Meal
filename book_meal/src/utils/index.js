import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}