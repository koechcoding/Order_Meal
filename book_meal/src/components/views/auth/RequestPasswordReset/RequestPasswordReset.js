import React from 'react';
import axios from 'src/axios';
import { Button, Alert } from 'reactstrap';
import { Redirect } from 'react-router-dom';
import { Role } from 'src/constants';

import AuthPage from 'src/components/common/AuthPage';
import { singleError, authenticated } from 'src/utils';

class RequestPasswordReset extends React.Component {
    state={}

    onSubmit=(e)=>{
        e.preventDefault()
        this.setState({
            ...this.state,
            loading: true,
        });
        axios.post('auth/password-reset', this.state).then(({ data }) =>{
            this.setState({
                ...this.state,
                success: true,
                loading: false,
            });
        });
    }

    onChange=(e)=>{
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        });
    }

    render(){
        const { error, success, loading } = this.state
        const user = authenticated();
        if(user){
            switch(user.role){
                case Role.SUPER_ADMIN:
                case Role.SUPER_ADMIN:
                    return <Redirect to='/admin/meals' />
                case Role.USER:
                    return <Redirect to="/user/menu" />
                default:
                    break;
            }
        }

        return(
            <AuthPage styles={{ minHeight: '500px', height: '100vh'}} loading={loading}>
                <form
                    className='col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card'
                    onSubmit={this.onSubmit}>
                    <h5 className='text-center mb-3 mt-4'>Request Password Reset</h5>
                    {error &&
                        <Alert color="success">
                            An email has been sent with the password reset link.
                            Please login to this email account to proceed.
                        </Alert>
                    }
                    <label>Account Email Address</label>
                    <input  
                       type="email"
                       className='form-control'
                       name="email"
                       onChange={this.onChange} />
                    <Button disabled={loading} className="btn btn-primary mt-2 mb-4">Reset</Button>
                </form>
            </AuthPage>
        );
    }
} 

export default RequestPasswordReset;