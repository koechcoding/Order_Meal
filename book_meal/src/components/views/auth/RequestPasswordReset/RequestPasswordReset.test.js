import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; // order matters!
import RequestPasswordReset from './RequestPasswordReset';

describe('<RequestPasswordReset />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<RequestPasswordReset/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('records changes', () => {
        const wrapper = shallow(<RequestPasswordReset />);
        wrapper.instance().onChange({target:{name: 'name', value: 'value'}});
        expect(wrapper.state('name')).toBe('value');
    });
});