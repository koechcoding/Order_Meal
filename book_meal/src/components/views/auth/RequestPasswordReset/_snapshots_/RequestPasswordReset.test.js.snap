// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestPasswordReset /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestPasswordReset />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card"
        onSubmit={[Function]}
>
        <h5
                className="text-center mb-3 mt-4"
        >
                Request Password Reset
        </h5>
        <label>
                Account Email Address
        </label>
        <input
                className="form-control"
                name="email"
                onChange={[Function]}
                type="email"
        />
        <Button
                className="btn btn-primary mt-2 mb-4"
                color="secondary"
                disabled={undefined}
                tag="button"
        >
                Reset
        </Button>
</form>,
      "loading": undefined,
      "styles": Object {
        "height": "100vh",
        "minHeight": "500px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="text-center mb-3 mt-4"
>
            Request Password Reset
</h5>,
          undefined,
          undefined,
          <label>
            Account Email Address
</label>,
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            type="email"
/>,
          <Button
            className="btn btn-primary mt-2 mb-4"
            color="secondary"
            disabled={undefined}
            tag="button"
>
            Reset
</Button>,
        ],
        "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Request Password Reset",
            "className": "text-center mb-3 mt-4",
          },
          "ref": null,
          "rendered": "Request Password Reset",
          "type": "h5",
        },
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Account Email Address",
          },
          "ref": null,
          "rendered": "Account Email Address",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "email",
            "onChange": [Function],
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Reset",
            "className": "btn btn-primary mt-2 mb-4",
            "color": "secondary",
            "disabled": undefined,
            "tag": "button",
          },
          "ref": null,
          "rendered": "Reset",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card"
          onSubmit={[Function]}
>
          <h5
                    className="text-center mb-3 mt-4"
          >
                    Request Password Reset
          </h5>
          <label>
                    Account Email Address
          </label>
          <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    type="email"
          />
          <Button
                    className="btn btn-primary mt-2 mb-4"
                    color="secondary"
                    disabled={undefined}
                    tag="button"
          >
                    Reset
          </Button>
</form>,
        "loading": undefined,
        "styles": Object {
          "height": "100vh",
          "minHeight": "500px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="text-center mb-3 mt-4"
>
              Request Password Reset
</h5>,
            undefined,
            undefined,
            <label>
              Account Email Address
</label>,
            <input
              className="form-control"
              name="email"
              onChange={[Function]}
              type="email"
/>,
            <Button
              className="btn btn-primary mt-2 mb-4"
              color="secondary"
              disabled={undefined}
              tag="button"
>
              Reset
</Button>,
          ],
          "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Request Password Reset",
              "className": "text-center mb-3 mt-4",
            },
            "ref": null,
            "rendered": "Request Password Reset",
            "type": "h5",
          },
          undefined,
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Account Email Address",
            },
            "ref": null,
            "rendered": "Account Email Address",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "email",
              "onChange": [Function],
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Reset",
              "className": "btn btn-primary mt-2 mb-4",
              "color": "secondary",
              "disabled": undefined,
              "tag": "button",
            },
            "ref": null,
            "rendered": "Reset",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;